src/skadi/sample.cc

```diff
@@ -71,14 +72,14 @@ uint64_t file_size(const std::string& file_name) {
 namespace valhalla {
 namespace skadi {
 
-enum class format_t { UNKNOWN = 0, RAW = 1, GZIP = 2 };
+enum class format_t { UNKNOWN = 0, RAW = 1, GZIP = 2, LZ4 = 3 };
 
 class cache_item_t {
 private:
   format_t format;
   valhalla::midgard::mem_map<char> data;
   int usages;
-  const char* unpacked;
```

Comment:
this is unfortunate but i assume its an artifact of `LZ4F_decompress` taking a non const parameter

Reply:
the good news is you dont need all of these removals of `const` because when we create the bytes where we unpack the data, we create it as non-const. so we can const cast it back before you make that call. ill make a suggestion below and then we can remove a lot of the changes here

Reply:
Ah, great! I cleaned it all up as suggested.